clients/java/signalr/src/main/java/HubConnection.java

```diff
@@ -15,21 +15,38 @@
     private CallbackMap handlers = new CallbackMap();
     private HubProtocol protocol;
     private Gson gson = new Gson();
+    private Boolean handshakeReceived = false;
+    private static final String RECORD_SEPARATOR = "\u001e";
     private HubConnectionState connectionState = HubConnectionState.DISCONNECTED;
 
+    // Only json is supported currently
+    private String protocolName = "json";
```

Comment:
These belong on the protocol class, see https://github.com/aspnet/SignalR/blob/release/2.2/clients/ts/signalr/src/JsonHubProtocol.ts#L16-L18